pyqldb/driver/qldb_driver.py

```diff
@@ -10,22 +10,21 @@
 # and limitations under the License.
 from queue import Queue, Empty
 from logging import getLogger
+from time import sleep
 from threading import BoundedSemaphore
 from warnings import warn
 
 from boto3 import client
 from boto3.session import Session
 from botocore.config import Config
-from botocore.exceptions import ClientError
-from pyqldb.config.retry_config import RetryConfig
 
 from .. import __version__
 from ..communication.session_client import SessionClient
-
-from ..errors import DriverClosedError, SessionPoolEmptyError, is_invalid_session_exception, \
-    is_transaction_expired_exception
+from ..config.retry_config import RetryConfig
+from ..errors import ExecuteError, DriverClosedError, SessionPoolEmptyError
 from ..session.qldb_session import QldbSession
 from ..util.atomic_integer import AtomicInteger
+from ..util.retry import Retry
 
 POOL_TIMEOUT_SECONDS = 0.001
 logger = getLogger(__name__)
```

Comment:
For the close function, let's do a while true loop until the method raises Empty, catch it and return.

That makes it more threadsafe.

Reply:
Updated